packages/now-next/src/utils.ts

```diff
@@ -490,7 +495,7 @@ export async function getDynamicRoutes(
       routes.push({
         src: pageMatcher.matcher.source,
         dest,
-        check: true,
+        check: !isDev,
```

Comment:
Is this change needed? Seems like it shouldn't affect the routes in dev mode 🤔

Reply:
Actually that change _is_ the fix, all the rest was just propagating the value of `isDev` though the function calls because I failed to notice that getRoutes was always only called in dev mode. 

I initially added tons of traces in the route matching code and ended up with traces like 
```
dev router trying to match  undefined /blog/top-10-des-incontournables-destination-du-perou 
dev router trying to match  GET /www/blog/top-10-des-incontournables-destination-du-perou to {"src":"^/www/(blog/([^/]+?)(?:/)?)$","dest":"http://localhost:44659/$1","check":true}
===============
hasDestFile false pathname /blog/top-10-des-incontournables-destination-du-perou destPath http://localhost:44659/blog/top-10-des-incontournables-destination-du-perou phase null routeConfig {"src":"^/www/(blog/([^/]+?)(?:/)?)$","dest":"http://localhost:44659/$1","check":true}
===============
```
After a bit of back and forth testing a few vercel.json configurations @styfle wrote: 
```
Oh so the route `
{"src":"^/www/(blog/([^/]+?)(?:/)?)$","dest":"http://localhost:43097/$1","check":true}
is from @vercel/next which seems wrong. I dont think it should have check:true
```
I traced it back to this line and indeed changing it to false resolved the issue. I must confess that I have absolutely *no idea* what `check: true` or`check: false` does. since I needed is to be false in dev to get the behaviour I wanted, I made the safest possible change I could by making sure this only affected dev mode.

on a side note, since the value of the `isDev` parameter was not provided to `getDynamicRoutes` before this change `isDev` was always undefined, thus `!isDev` was always true, thus the code at https://github.com/vercel/vercel/pull/4510/files#diff-f5c41b908e3f767349246f7dfb3714e1L484 was always going through `path.join('/', entryDirectory, pageMatcher.pageName)`  but I was unable to detect an impact after changing it.

Reply:
@ijjk The problem is that `check: true` is similar to `continue: true` when the resource is not found. See the routing logic here: https://github.com/vercel/vercel/blob/d7731d191b0ab808a5b3f25a73349e891857b017/packages/now-cli/src/util/dev/router.ts#L120

In particular, this is where it continues: https://github.com/vercel/vercel/blob/d7731d191b0ab808a5b3f25a73349e891857b017/packages/now-cli/src/util/dev/router.ts#L149